### PR TITLE
feature: add support for unmanaged and remove for servicemesh and serverless

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

ref https://issues.redhat.com/browse/RHOAIENG-443

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
Looks like this needs update in unit tests as well?

**Comment by zdtsw** [comment]
> Looks like this needs update in unit tests as well?

yep, i am fixing the tests

**Comment by bartoszmajsak** [comment]
Shouldn't we also extend CRD through kubebuilder directives? We are not listing all the values there.

**Comment by zdtsw** [comment]
> Shouldn't we also extend CRD through kubebuilder directives? We are not listing all the values there.

if you meant https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/infrastructure/v1/serverless_types.go#L10 and https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/infrastructure/v1/servicemesh_types.go#L7 they had been done in downstream first and get in ODH after backport

File: components/kserve/kserve.go [hunk 1]
```diff
@@ -158,11 +158,22 @@ func (k *Kserve) Cleanup(_ client.Client, instance *dsciv1.DSCInitializationSpec
...
+		if err := k.removeServerlessFeatures(instance); err != nil {
+			return err
 		}
 
+	case operatorv1.Managed: // standard workflow to create CR
+		switch instance.ServiceMesh.ManagementState {
+		case operatorv1.Unmanaged, operatorv1.Removed:
+			return fmt.Errorf("ServiceMesh is need to set to 'Managaed' in DSCI CR, it is required by KServe serving field")
+		}
+		// create serving
```
Comment (bartoszmajsak) [inline components/kserve/kserve.go:176]: Not sure why this comment here helps?

```suggestion
```

**Review by bartoszmajsak — APPROVED** [review]
Just one nit. I wasn't aware CRD changes came with downstream backport.

**Pull request merged**